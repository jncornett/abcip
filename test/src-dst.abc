# applies to all subsequent a and b packets
c( eth:a="f:e:d:c:b:a", b="9:8:7:6:5:4";
   ip4:a="192.168.1.2", b="192.168.1.3";
   tcp:a=46240,         b=80 )

a(syn)
b(syn, ack)
a(ack)

d ( stack="eth:ip6:udp" )

# applies to all subsequent a packets
a ( ip6:src="::1", dst="::2"; tcp:src=3, dst=4, len=16 )
b ( len=16 )
a ( len=16 )

# applies to all subsequent b packets
b ( ip6:src="::6", dst="::7"; 4:src=8, dst=9, len=16 )
a ( len=16 )
b ( len=16 )

