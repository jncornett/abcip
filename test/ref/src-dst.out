IP (tos 0x0, ttl 64, id 1, offset 0, flags [none], proto TCP (6), length 40)
    192.168.1.2.46240 > 192.168.1.3.80: Flags [S], cksum 0x579b (correct), seq 1, win 8192, length 0
	0x0000:  0908 0706 0504 0f0e 0d0c 0b0a 0800 4500  ..............E.
	0x0010:  0028 0001 0000 4006 f779 c0a8 0102 c0a8  .(....@..y......
	0x0020:  0103 b4a0 0050 0000 0001 0000 0000 5002  .....P........P.
	0x0030:  2000 579b 0000                           ..W...
IP (tos 0x0, ttl 64, id 1, offset 0, flags [none], proto TCP (6), length 40)
    192.168.1.3.80 > 192.168.1.2.46240: Flags [S.], cksum 0x5789 (correct), seq 1, ack 2, win 8192, length 0
	0x0000:  0f0e 0d0c 0b0a 0908 0706 0504 0800 4500  ..............E.
	0x0010:  0028 0001 0000 4006 f779 c0a8 0103 c0a8  .(....@..y......
	0x0020:  0102 0050 b4a0 0000 0001 0000 0002 5012  ...P..........P.
	0x0030:  2000 5789 0000                           ..W...
IP (tos 0x0, ttl 64, id 2, offset 0, flags [none], proto TCP (6), length 40)
    192.168.1.2.46240 > 192.168.1.3.80: Flags [.], cksum 0x578a (correct), ack 1, win 8192, length 0
	0x0000:  0908 0706 0504 0f0e 0d0c 0b0a 0800 4500  ..............E.
	0x0010:  0028 0002 0000 4006 f778 c0a8 0102 c0a8  .(....@..x......
	0x0020:  0103 b4a0 0050 0000 0002 0000 0002 5010  .....P........P.
	0x0030:  2000 578a 0000                           ..W...
IP6 (flowlabel 0x00001, hlim 64, next-header UDP (17) payload length: 24) ::1.3 > ::2.4: [udp sum ok] UDP, length 16
	0x0000:  0209 0807 0605 0201 0203 0405 86dd 6000  ..............`.
	0x0010:  0001 0018 1140 0000 0000 0000 0000 0000  .....@..........
	0x0020:  0000 0000 0001 0000 0000 0000 0000 0000  ................
	0x0030:  0000 0000 0002 0003 0004 0018 b8da 4c6f  ..............Lo
	0x0040:  7265 6d20 6970 7375 6d20 646f 6c6f       rem.ipsum.dolo
IP6 (flowlabel 0x00001, hlim 64, next-header UDP (17) payload length: 24) ::ffff:10.9.8.7.8 > ::ffff:10.1.2.3.48620: [udp sum ok] UDP, length 16
	0x0000:  0201 0203 0405 0209 0807 0605 86dd 6000  ..............`.
	0x0010:  0001 0018 1140 0000 0000 0000 0000 0000  .....@..........
	0x0020:  ffff 0a09 0807 0000 0000 0000 0000 0000  ................
	0x0030:  ffff 0a01 0203 0008 bdec 0018 dcdb 4c6f  ..............Lo
	0x0040:  7265 6d20 6970 7375 6d20 646f 6c6f       rem.ipsum.dolo
IP6 (flowlabel 0x00001, hlim 64, next-header UDP (17) payload length: 24) ::1.3 > ::2.4: [udp sum ok] UDP, length 16
	0x0000:  0209 0807 0605 0201 0203 0405 86dd 6000  ..............`.
	0x0010:  0001 0018 1140 0000 0000 0000 0000 0000  .....@..........
	0x0020:  0000 0000 0001 0000 0000 0000 0000 0000  ................
	0x0030:  0000 0000 0002 0003 0004 0018 e125 7220  .............%r.
	0x0040:  7369 7420 616d 6574 2c20 636f 6e73       sit.amet,.cons
IP6 (flowlabel 0x00001, hlim 64, next-header UDP (17) payload length: 24) ::6.8 > ::7.9: [udp sum ok] UDP, length 16
	0x0000:  0201 0203 0405 0209 0807 0605 86dd 6000  ..............`.
	0x0010:  0001 0018 1140 0000 0000 0000 0000 0000  .....@..........
	0x0020:  0000 0000 0006 0000 0000 0000 0000 0000  ................
	0x0030:  0000 0000 0007 0008 0009 0018 e111 7220  ..............r.
	0x0040:  7369 7420 616d 6574 2c20 636f 6e73       sit.amet,.cons
IP6 (flowlabel 0x00001, hlim 64, next-header UDP (17) payload length: 24) ::1.3 > ::2.4: [udp sum ok] UDP, length 16
	0x0000:  0209 0807 0605 0201 0203 0405 86dd 6000  ..............`.
	0x0010:  0001 0018 1140 0000 0000 0000 0000 0000  .....@..........
	0x0020:  0000 0000 0001 0000 0000 0000 0000 0000  ................
	0x0030:  0000 0000 0002 0003 0004 0018 a7a4 6563  ..............ec
	0x0040:  7465 7475 7220 6164 6970 6973 6369       tetur.adipisci
IP6 (flowlabel 0x00001, hlim 64, next-header UDP (17) payload length: 24) ::6.8 > ::7.9: [udp sum ok] UDP, length 16
	0x0000:  0201 0203 0405 0209 0807 0605 86dd 6000  ..............`.
	0x0010:  0001 0018 1140 0000 0000 0000 0000 0000  .....@..........
	0x0020:  0000 0000 0006 0000 0000 0000 0000 0000  ................
	0x0030:  0000 0000 0007 0008 0009 0018 a790 6563  ..............ec
	0x0040:  7465 7475 7220 6164 6970 6973 6369       tetur.adipisci
